app/models/hold.rb

```diff
@@ -19,22 +19,21 @@ class Hold < ActiveRecord::Base
   def handle_conflicting_rentals
     # if rental falls between data range of our hold
     # only reserved but not picked up
-    conflicting_rentals = Rental.reserved.where('start_time <= :hold_end_time AND end_time >= :hold_start_time AND item_id = :hold_item_id',
-                                                hold_start_time: start_time, hold_end_time: end_time, hold_item_id: item.id)
+    conflicting_rentals = Rental.reserved.joins('INNER JOIN rentals_items ON rentals_items.rental_id = rentals.id')
+                                .where('start_time <= :hold_end_time AND end_time >= :hold_start_time AND rentals_items.item_id = :hold_item_id',
```

Comment:
Interesting, I didn't know you could keyword-interpolate.

I'd suggest a model method for this.